android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java

```diff
@@ -645,6 +780,101 @@ void createTouchScalar(ImageResolution resolution, DisplayMetrics displayMetrics
         return motionEventList;
     }
 
+    public VideoStreamingParameters getLastCachedStreamingParameters() {
+        return parameters;
+    }
+
+    private List<VideoStreamingCapability> getSupportedCapabilities(
+            List<VideoStreamingRange> ranges,
+            List<VideoStreamingCapability> originalAdditionalCapabilities
+    ){
+
+        List<VideoStreamingCapability> validCapabilities = new ArrayList<>();
+
+        VideoStreamingCapability preferredCapability = new VideoStreamingCapability();
+        preferredCapability.setDiagonalScreenSize(parameters.getPreferredDiagonal());
+        preferredCapability.setPreferredResolution(new ImageResolution(
+                parameters.getResolution().getResolutionWidth(),
+                parameters.getResolution().getResolutionHeight())
+        );
+
+        // get the first one - the Desired resolution to guarantee streaming will start
+        validCapabilities.add(preferredCapability);
+        for (VideoStreamingRange range: ranges) {
+
+            Integer constraintHeightMax = range.getMaxResolution().getResolutionHeight();
+            Integer constraintHeightMin = range.getMinResolution().getResolutionHeight();
+
+            for (VideoStreamingCapability capability : originalAdditionalCapabilities) {
```

Comment:
We should add a null check for originalAdditionalCapabilities, it is possible for the HMI to send additionalVideoStreamingCapabilities as null and if that is the case there will be a NPE here